accounts/views.py

```diff
@@ -0,0 +1,63 @@
+from django.contrib.auth import get_user_model
+from rest_framework.generics import CreateAPIView, ListAPIView, GenericAPIView
+from rest_framework.response import Response
+from rest_framework.status import HTTP_200_OK, HTTP_400_BAD_REQUEST, HTTP_401_UNAUTHORIZED, HTTP_205_RESET_CONTENT, HTTP_400_BAD_REQUEST
+from rest_framework.views import APIView
+from django.contrib.auth import authenticate
+from rest_framework_simplejwt.tokens import RefreshToken
+from accounts.models import User
+from blog.permissions import IsAdminUser, IsLoggedIn
+from .serializers import UserSerializer, UserLoginSerializer
+
+
+class UserCreateAPIView(CreateAPIView):
+    """View to create a user"""
+    
+    serializer_class = UserSerializer
+
+
+class UserListAPIView(ListAPIView):
+    """View to get all user"""
+
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+
+class LoginAPIView(APIView):
+
+    def post(self, request):
+        
+        serializer = UserLoginSerializer(data=request.data)
+        
+        if serializer.is_valid():
+            email = serializer.data.get('email')
+            password = serializer.data.get('password')
+
+            if email is None or password is None:
+                return Response({'error': 'Please provide both email and password'}, status=HTTP_400_BAD_REQUEST)
+
+            user = User.objects.get(email=email)
```

Comment:
you should get `user` after you've checked if user exists or not. If you get user before any check the code will break. You've added `if` condition in line 41. So, you should get user with `.get()` after line 42 outside of `if` condition, this way your code will not break